tests/unittests/test_handler/test_handler_mounts.py

```diff
@@ -183,6 +183,45 @@ def device_name_to_device(self, path):
 
         return dev
 
+    @mock.patch('cloudinit.util.kernel_version')
+    @mock.patch('cloudinit.util.get_mount_info')
+    def test_swap_creation_method_xfs(
+            self, m_kernel_version, m_get_mount_info):
+        m_kernel_version.return_value = (3, 18)
+        m_get_mount_info.return_value = ["", "xfs"]
+        fstab = '/swap.img swap swap defaults 0 0\n'
+
+        with open(cc_mounts.FSTAB_PATH, 'w') as fd:
+            fd.write(fstab)
```

Comment:
As far as I understand, the code under test is responsible just to use the correct method for creating swap files. In order to test that we need to provide a mocked fstab with a swap line, filesystem type information and kernel version to see how it responds. The idea is to all these information the process it without any errors.

Reply:
`cc_mounts` writes out `/etc/fstab`: https://github.com/canonical/cloud-init/blob/411dbbebd328163bcb1c676cc711f3e5ed805375/cloudinit/config/cc_mounts.py#L520-L521

I think the only reason to write to it in tests is if you are testing the logic which preserves existing entries; I don't believe these tests need to check that.

I would expect these tests to, after calling `cc_mounts.handle` assert that (a) the correct swapfile creation command (or commands) are executed, and that (b) `swapon -a` is called. The absence of an error is not an indication that we have successfully created a swapfile, particularly because we have a few exception handlers which (appropriately, I think) swallow exceptions, e.g.: https://github.com/canonical/cloud-init/blob/411dbbebd328163bcb1c676cc711f3e5ed805375/cloudinit/config/cc_mounts.py#L352-L353

Also, now I'm looking at the other tests here, this is in a test class named `TestFstabHandling`; these tests are not testing fstab handling, so I think they would be best moved into their own new class (which would also let you make them pytest tests if you would like).